Lisvindanu/kelompok3_romusha
# Implement RBAC, View composer, and caching for dynamic navbar and user access control

### Changes
- Implemented RBAC to restrict access to admin-only pages.
- Added caching for user data retrieval to improve performance.
- Updated `AppServiceProvider` to use View composer for sharing user data across views.
- Navbar now dynamically adjusts based on user roles (e.g., admin or regular user).

### Technical Details
- Used `View::composer` in `AppServiceProvider` to share user data dynamically.
- Caching implemented with `Cache::remember` for efficient user data fetching.
- Handled invalid tokens by clearing sessions to maintain security.

### Impact
- Improved security by restricting unauthorized access.
- Enhanced performance by reducing repetitive API calls with caching.
- Better UX with dynamic navbar adjustments.

### How to Test
1. Log in as a regular user and verify that the navbar shows options relevant to users.
2. Log in as an admin and ensure admin-specific pages are accessible.
3. Check that invalid tokens are handled properly (e.g., session cleared when token expires).
4. Verify that user data is cached for improved performance.
